roles/os_firewall/tasks/iptables.yml

```diff
@@ -4,17 +4,28 @@
   systemd:
     name: firewalld
     state: stopped
-    enabled: no
-    masked: yes
   register: task_result
   failed_when:
     - task_result is failed
     - ('could not' not in task_result.msg|lower)
 
+# These tasks are split in two, as changing masked and enabled status
+# would mutate `changed` field, so it needs to be performed in two tasks
+- name: Mask and disable firewalld service
+  systemd:
+    name: firewalld
+    enabled: no
+    masked: yes
+  when:
+    - task_result is succeeded
+    - task_result.failed_when_result
```

Comment:
This will never be true, otherwise we would have failed the previous task.

Reply:
Fixed